### PR TITLE
Add moser-de bruijin sequence

### PR DESCRIPTION
**Pull request by cp-sumi-k**

### CODE REVIEW

File: math/moserdebruijnsequence/sequence.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// The Moser-de Bruijn sequence is the sequence obtained by
+// adding up the distinct powers of the number 4 (For example 1, 4, 16, 64, etc).
```
Comment (Navid2zp) [inline math/moserdebruijnsequence/sequence.go:2]: You should probably add the Wikipedia URL:
https://en.wikipedia.org/wiki/Moser%E2%80%93de_Bruijn_sequence

File: math/moserdebruijnsequence/sequence.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+package main
+
+import "fmt"
+
+func main() {
```
Comment (Navid2zp) [inline math/moserdebruijnsequence/sequence.go:8]: Also, you should remove the main function.
You can change your `getSequenceNumber` function to `MoserDeBruijnSequence` instead and return the final result.

**Review by Navid2zp — CHANGES_REQUESTED** [review]
You need to add the tests too. Check other algorithms for example.

**Comment by cp-sumi-k** [comment]
Changes Done @Navid2zp. Please review it.

**Comment by Navid2zp** [comment]
> Changes Done @Navid2zp. Please review it.

I'm not a maintainer but looks good to me. Wait for one of the maintainers to check.
Just wanted to point out the simple problems so you can fix it before one of the maintainers check the PR since it usually takes a while.

**Review by Navid2zp — DISMISSED** [review]

**Review by iamrajiv — DISMISSED** [review]

File: math/moserdebruijnsequence/sequence_test.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+)
+
+var testCases = []struct {
+	name     string
+	inputNum int
+	expected []int
+}{
+	{"first 0 terms", 0, []int{}},
+	{"first 5 terms", 5, []int{0, 1, 4, 5, 16}},
+	{"first 10 terms", 10, []int{0, 1, 4, 5, 16, 17, 20, 21, 64, 65}},
```
Comment (siriak) [inline math/moserdebruijnsequence/sequence_test.go:15]: No need to test the first 0, 5, and 10 terms if you're testing the first 15 terms too

**Review by siriak — APPROVED** [review]
LGTM except for 1 comment below. Please bring the branch up to date so that I could merge it in.

**Review by siriak — APPROVED** [review]

**Pull request merged**